python/nav/bin/update_ouis.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+# -*- testargs: -h -*-
+#
+# Copyright (C) 2024 Sikt
+#
+# This file is part of Network Administration Visualized (NAV).
+#
+# NAV is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License version 3 as published by
+# the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.  You should have received a copy of the GNU General Public License
+# along with NAV. If not, see <http://www.gnu.org/licenses/>.
+#
+
+import argparse
+import sys
+import logging
+
+from typing import Iterable, Generator, Tuple
+
+import requests
+from requests.exceptions import RequestException
+
+from nav.bootstrap import bootstrap_django
+
+bootstrap_django(__file__)
+
+import django.db
+
+from nav.macaddress import MacPrefix
+from nav.logs import init_stderr_logging
+
+_logger = logging.getLogger(__name__)
+
+FILE_URL = "https://standards-oui.ieee.org/oui/oui.txt"
+
+MAX_ERRORS = 100
+
+
+def main():
+    init_stderr_logging()
+    argparse.ArgumentParser(
+        description="Updates the database with OUIs and their related organizations"
+    ).parse_args()
+    run()
+
+
+def run():
+    _logger.debug(f"Downloading OUIs from {FILE_URL}")
+    try:
+        data = _download_oui_file(FILE_URL)
+    except RequestException as error:
+        _logger.error("Error while downloading OUIs: %s", error)
+        sys.exit(1)
+    ouis = _parse_ouis(data)
+    _update_database(ouis)
+    _logger.debug("Completed updating OUI records")
+
+
+def _download_oui_file(url: str) -> str:
+    response = requests.get(url)
+    response.raise_for_status()
+    return response.text
+
+
+def _parse_ouis(oui_data: str) -> Generator[Tuple[str, str], None, None]:
+    """Returns lists of tuples containing OUI and vendor name for
+    each vendor
+    """
+    error_count = 0
+    for line in oui_data.split('\n'):
+        if "(hex)" not in line:
+            continue
+        try:
+            yield _parse_line(line)
+        except ValueError as error:
+            _logger.error(
+                "Attempting to parse the line '%s' gave error: %s", line, error
+            )
```

Comment:
Here you are logging per error until it reaches the max error amount

Reply:
lol must have lost that commit along the way

Reply:
There, had just forgotten to push it